phoenix-core/src/it/java/org/apache/phoenix/execute/PartialCommitIT.java

```diff
@@ -0,0 +1,262 @@
+    
+    private static final String TABLE_NAME_TO_FAIL = "b_failure_table".toUpperCase();
+    private static final byte[] ROW_TO_FAIL = Bytes.toBytes("fail me");
+    private static final String UPSERT_TO_FAIL = "upsert into " + TABLE_NAME_TO_FAIL + " values ('" + Bytes.toString(ROW_TO_FAIL) + "', 'boom!')";
+    private static final String DELETE_TO_FAIL = "delete from " + TABLE_NAME_TO_FAIL + "  where k='z'";
+    private static final HBaseTestingUtility TEST_UTIL = new HBaseTestingUtility();
+    private static String url;
+    private static Driver driver;
+    private static final Properties props = new Properties();
+    
+    static {
+        props.put(PhoenixRuntime.CURRENT_SCN_ATTRIB, 10);
+    }
+    
+    @BeforeClass
+    public static void setupCluster() throws Exception {
+      Configuration conf = TEST_UTIL.getConfiguration();
+      setUpConfigForMiniCluster(conf);
+      conf.setClass("hbase.coprocessor.region.classes", FailingRegionObserver.class, RegionObserver.class);
+      conf.setBoolean("hbase.coprocessor.abortonerror", false);
+      conf.setBoolean(Indexer.CHECK_VERSION_CONF_KEY, false);
+      TEST_UTIL.startMiniCluster();
+      String clientPort = TEST_UTIL.getConfiguration().get(QueryServices.ZOOKEEPER_PORT_ATTRIB);
+      url = JDBC_PROTOCOL + JDBC_PROTOCOL_SEPARATOR + LOCALHOST + JDBC_PROTOCOL_SEPARATOR + clientPort
+              + JDBC_PROTOCOL_TERMINATOR + PHOENIX_TEST_DRIVER_URL_PARAM;
+
+      Map<String, String> props = Maps.newHashMapWithExpectedSize(1);
+      // Must update config before starting server
+      props.put(QueryServices.DROP_METADATA_ATTRIB, Boolean.toString(true));
+      driver = initAndRegisterDriver(url, new ReadOnlyProps(props.entrySet().iterator()));
+      createTablesWithABitOfData();
+    }
+    
+    private static void createTablesWithABitOfData() throws Exception {
+        Properties props = new Properties();
+        props.put(PhoenixRuntime.CURRENT_SCN_ATTRIB, 10);
+
+        try (Connection con = driver.connect(url, new Properties())) {
+            Statement sta = con.createStatement();
+            sta.execute("create table a_success_table (k varchar primary key, c varchar)");
+            sta.execute("create table b_failure_table (k varchar primary key, c varchar)");
+            sta.execute("create table c_success_table (k varchar primary key, c varchar)");
+            con.commit();
+        }
+
+        props.put(PhoenixRuntime.CURRENT_SCN_ATTRIB, 100);
+
+        try (Connection con = driver.connect(url, new Properties())) {
+            con.setAutoCommit(false);
+            Statement sta = con.createStatement();
+            for (String table : newHashSet("a_success_table", TABLE_NAME_TO_FAIL, "c_success_table")) {
+                sta.execute("upsert into " + table + " values ('z', 'z')");
+                sta.execute("upsert into " + table + " values ('zz', 'zz')");
+                sta.execute("upsert into " + table + " values ('zzz', 'zzz')");
```

Comment:
Since callers supply bind variables, they can keep track of them if they are interested in handling CommitException.getUncommittedStatementIndexes() correctly. e.g.:
```
try {
    PreparedStatement ps = con.prepareStatement("upsert into t values (?,?)");
    
    ps.setString(1, "a");
    ps.setString(2, "a");
    ps.executeUpdate(); // statement index 0
    
    ps.setString(1, "b");
    ps.setString(2, "b");
    ps.executeUpdate(); // statement index 1

    con.commit();
} catch (CommitException ce) {
    Set<Integer> uncommitedStatementIndexes = ce.getUncommittedStatementIndexes();
    // do something with uncommitedStatementIndexes 
}
```